lib/buildFile.js

```diff
@@ -42,8 +45,49 @@ function buildFile(destination, format, platform, dictionary, filter) {
   if (!fs.existsSync(dirname))
     fs.mkdirsSync(dirname);
 
-  fs.writeFileSync(destination, format(filterProperties(dictionary, filter), platform));
-  console.log(chalk.bold.green('✔︎') + ' ' + destination);
+  let filteredProperties = filterProperties(dictionary, filter);
+
+  // Check for property name Collisions
+  var nameCollisionObj = {};
+  filteredProperties.allProperties && filteredProperties.allProperties.forEach((propertyData) => {
```

Comment:
Do we need this check? I think when we are constructing `allProperties` it will always be at least an empty array?

Reply:
Unfortunately this is required. The tests do not generate allProperties at all in some cases, causing an error to throw here and the tests to fail.